samples/nrf9160/modem_shell/src/main.c

```diff
@@ -161,4 +161,9 @@ void main(void)
 #if defined(CONFIG_BOOTLOADER_MCUBOOT)
 	boot_write_img_confirmed();
 #endif
+
+	/* Resize terminal width and height of the shell to have proper command editing. */
+	shell_execute_cmd(shell_global, "resize");
+	/* Run empty command because otherwise "resize" would be set to the command line. */
+	shell_execute_cmd(shell_global, "");
```

Comment:
This looks like a workaround for a bug in the shell implementation. Is there a bug for this? Looking at shell.c, `shell_execute_cmd()` calls `execute()` to execute the command. The same `execute()` function is also used to process commands from the command line, however in that case there's also `state_set(shell, SHELL_STATE_ACTIVE)` call which clears the command by calling `cmd_buffer_clear()`. I think something like this should happen also after `shell_execute_cmd()` has called `execute()` to get rid of the already processed command.

Reply:
No bug for shell at least yet.
@rlubos, what do you think we should do here? Is this something that should be fixed to zephyr side?

Reply:
@jakub-uC Could you comment this behavior from Zephyr shell perspective, i.e., should this be fixed in there or require this "hack" in the clients?

Reply:
I'll resolve this issue with Jakub to see if Zephyr needs a fix, and then remove it from here but for now, I guess we could approve the changes to get them to the master.

Reply:
@trantanen @tokangas 
It is not a bug it is a feature :)
Function `shell_execute_cmd` was implemented only for internal usage like unit tests. However, your use case is very interesting and I believe it is a good idea to implement it in the upstream.

Reply:
@jakub-uC Thanks for handling this!

Reply:
I'll handle MOSH side change when Zephyr is done with ticket MOSH-148

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/37548